src/table_builder.cc

```diff
@@ -55,16 +56,17 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
              value.size() >= cf_options_.min_blob_size &&
              cf_options_.blob_run_mode == TitanBlobRunMode::kNormal) {
     // we write to blob file and insert index
-    std::string index_value;
-    AddBlob(ikey.user_key, value, &index_value);
+    BlobRecord record;
+    record.key = ikey.user_key;
+    record.value = value;
+    BlobFileBuilder::OutContexts contexts;
+    AddBlob(record, ikey, &contexts);
+    AddToBaseTable(contexts);
```

Comment:
if it's in buffer state, the blob record wouldn't be written into SST now. Whereas, for the next key-value which is of no need to write into Titan, it will be written into SST immediately. At last, the buffered blob records will be written into SST at `Finish()`. In this case, the SST is out of order.

Reply:
might need to copy the value twice to cachedCtx and outCtx to solve this.

Reply:
might need to copy the value twice to out-context and cached-context to solve this.

Reply:
I think we should introduce buffer state at table builder level.

Reply:
resolved in this commit:  [9073ecca81164f89f36c412e44eac197b80f7197](https://github.com/tikv/titan/pull/189/commits/9073ecca81164f89f36c412e44eac197b80f7197)

discussed here: https://tikv-wg.slack.com/archives/CQ545HDGC/p1602303591032200?thread_ts=1601073597.011400&cid=CQ545HDGC